src/components/views/dialogs/HostSignupDialog.tsx

```diff
@@ -0,0 +1,266 @@
+            },
+        );
+    }
+
+    public componentDidMount() {
+        window.addEventListener("message", this.messageHandler);
+    }
+
+    public componentWillUnmount() {
+        if (HostSignupStore.instance.isHostSignupActive) {
+            // Run the close dialog actions if we're still active, otherwise good to go
+            return this.closeDialog();
+        }
+    }
+
+    public render(): React.ReactNode {
+        return (
+            <div className="mx_HostSignup_persisted">
+                <PersistedElement key="host_signup" persistKey="host_signup">
+                    <div className={this.state.minimized ? "" : "mx_Dialog_wrapper"}>
+                        <div className={["mx_Dialog",
+                            this.state.minimized ? "mx_HostSignupDialog_minimized" : "mx_HostSignupDialog"].join(" ")
+                        }>
+                            {this.state.minimized &&
+                                <div className="mx_Dialog_header mx_Dialog_headerWithButton">
+                                    <div className="mx_Dialog_title">
+                                        {this.config.minimizedDialogTitle}
+                                    </div>
+                                    <AccessibleButton
+                                        className="mx_HostSignup_maximize_button"
+                                        onClick={this.maximizeDialog}
+                                        aria-label={_t("Maximize dialog")}
+                                        title={_t("Maximize dialog")}
+                                    />
+                                </div>
+                            }
+                            {!this.state.minimized &&
+                                <div className="mx_Dialog_header mx_Dialog_headerWithCancel">
+                                    <AccessibleButton
+                                        onClick={this.minimizeDialog} className="mx_HostSignup_minimize_button"
+                                        aria-label={_t("Minimize dialog")}
+                                        title={_t("Minimize dialog")}
+                                    />
+                                    <AccessibleButton
+                                        onClick={this.onCloseClick} className="mx_Dialog_cancelButton"
+                                        aria-label={_t("Close dialog")}
+                                        title={_t("Close dialog")}
+                                    />
+                                </div>
+                            }
+                            {this.state.error &&
+                                <div>
+                                    {this.state.error}
+                                </div>
+                            }
+                            {!this.state.error &&
+                                <iframe
+                                    src={this.config.url}
```

Comment:
surely we should be sending the user's language to the iframe for i18n?

Reply:
In our use case this wont be needed, but can add for other use cases :+1:

Reply:
So will Element Pro be for English speakers only?

Reply:
Currently the EMS host setup is only available in English. That could change in the future. But of course another provider might make use of this.

Lets avoid going into EMS or Element Pro design specifics in this issue though.

Reply:
Is the concern that some strings will be translated and some not? Would it be better to move all the strings inside the setup flow into configuration, for better control and to avoid partially translated screens?

Reply:
The concern is that very dangerous actions, like sharing an access token, may be achieved without the user being able to understand their actions due to their broken understanding of english and lack of translations

Reply:
This was resolved by making the strings translatable.